### PR TITLE
Update to bazel 3.4.1

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Bring us to the latest bazel version.

 * Prepares the ground for `bazel coverage` (code coverage via bazel with CI caches)
 * Make it possible to use the bazel3 arm releases which are now happening officially on bazel

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhlhahaha** [comment]
Hi Roman, I will test it on ARM.

**Review by dhiller — APPROVED** [review]
/lgtm
/retest

**Comment by xpivarc** [comment]
/hold

`make generate` won't pass

**Comment by rmohr** [comment]
/hold cancel

Rebased and updated. @dhiller @xpivarc could you have another look?

**Review by dhiller — APPROVED** [review]
/lgtm
/approve

**Comment by rmohr** [comment]
Needed to run `make generate`.

**Review by dhiller — APPROVED** [review]
/lgtm

File: hack/builder/version.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
-VERSION=30-7.0.0
+VERSION=30-8.0.2
```
Comment (dhiller) [inline hack/builder/version.sh:1]: Quick question: I never found out what versioning schema we are using for this?

Comment (rmohr) [reply hack/builder/version.sh:1]: There is none. I just bumped from 7 to 8. But there were various rebases, so I then changed the patch level to 1 and then 2 but I really just did something.

**Review by dhiller — APPROVED** [review]
/lgtm

**Pull request merged**